src/game.zig

```diff
@@ -32,45 +33,42 @@ const physics_system = @import("systems/physics_system.zig");
 const terrain_quad_tree_system = @import("systems/terrain_quad_tree.zig");
 const patch_prop_system = @import("systems/patch_prop_system.zig");
 const procmesh_system = @import("systems/procedural_mesh_system.zig");
+const static_mesh_renderer_system = @import("systems/static_mesh_renderer_system.zig");
 const state_machine_system = @import("systems/state_machine_system.zig");
 const timeline_system = @import("systems/timeline_system.zig");
-// const terrain_system = @import("systems/terrain_system.zig");
 // const gui_system = @import("systems/gui_system.zig");
 
 const SpawnContext = struct {
     flecs_world: *flecs.World,
     physics_world: *zphy.PhysicsSystem,
+    prefab_manager: *pm.PrefabManager,
     // player_pos: [3]f32,
 };
 
+var spider_prefab: flecs.Entity = undefined;
+var bow_prefab: flecs.Entity = undefined;
+
 fn spawnSpider(entity: flecs.EntityId, data: *anyopaque) void {
     _ = entity;
     const ctx = util.castOpaque(SpawnContext, data);
+
     const player_ent_id = flecs.c.ecs_lookup(ctx.flecs_world.world, "player");
     const player_ent = flecs.Entity{ .id = player_ent_id, .world = ctx.flecs_world.world };
     const player_pos = player_ent.get(fd.Position).?;
 
-    var ent = ctx.flecs_world.newEntity();
+    var ent = ctx.prefab_manager.instantiatePrefab(ctx.flecs_world, spider_prefab);
+
     var spawn_pos = [3]f32{
         player_pos.x + 10,
         player_pos.y + 2,
         player_pos.z + 10,
     };
-    ent.set(fd.Position{
+    ent.setOverride(fd.Position{
```

Comment:
Is setOverride really correct here? I thought that was supposed to be set on the prefab

Reply:
Yes, I think you're right. I've fixed all uses now